homeassistant/components/rainbird/__init__.py

```diff
@@ -36,9 +35,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     hass.data.setdefault(DOMAIN, {})
 
+    clientsession = async_create_clientsession()
+    entry.async_on_unload(clientsession.close)
```

Comment:
Probably close the session on Home Assistant stop too.

Reply:
I considered this, but intentionally left this out because it seemed like a lot of code with unclear benefit to me. I think it makes sense for a connection pool with thousands of connections, but for a single connection it seemed fine to just wait for unload? I've sent PR #112292 however since you've asked for it.

Reply:
What do you mean "to just wait for unload"?